pay-api/src/pay_api/services/eft_short_names.py

```diff
@@ -231,15 +231,19 @@ def _apply_payment_action(cls, short_name_id: int, auth_account_id: str):
     @classmethod
     def _get_statement_credit_invoice_links(cls, shortname_id, statement_id) -> List[EFTCreditInvoiceLinkModel]:
         """Get most recent EFT Credit invoice links associated to a statement and short name."""
-        return (db.session.query(EFTCreditInvoiceLinkModel)
-                .distinct(EFTCreditInvoiceLinkModel.invoice_id)
-                .join(EFTCreditModel, EFTCreditModel.id == EFTCreditInvoiceLinkModel.eft_credit_id)
-                .join(StatementInvoicesModel, StatementInvoicesModel.invoice_id == EFTCreditInvoiceLinkModel.invoice_id)
-                .filter(StatementInvoicesModel.statement_id == statement_id)
-                .filter(EFTCreditModel.short_name_id == shortname_id)
-                .filter(EFTCreditInvoiceLinkModel.status_code != EFTCreditInvoiceStatus.CANCELLED.value)
-                .order_by(EFTCreditInvoiceLinkModel.invoice_id, EFTCreditInvoiceLinkModel.created_on.desc())
-                ).all()
+        query = (db.session.query(EFTCreditInvoiceLinkModel)
+                 .distinct(EFTCreditInvoiceLinkModel.invoice_id)
+                 .join(EFTCreditModel, EFTCreditModel.id == EFTCreditInvoiceLinkModel.eft_credit_id)
+                 .join(StatementInvoicesModel,
+                       StatementInvoicesModel.invoice_id == EFTCreditInvoiceLinkModel.invoice_id)
+                 .filter(StatementInvoicesModel.statement_id == statement_id)
+                 .filter(EFTCreditModel.short_name_id == shortname_id)
+                 .filter(EFTCreditInvoiceLinkModel.status_code != EFTCreditInvoiceStatus.CANCELLED.value)
+                 .order_by(EFTCreditInvoiceLinkModel.invoice_id.desc(),
+                           EFTCreditInvoiceLinkModel.created_on.desc(),
+                           EFTCreditInvoiceLinkModel.id.desc())
```

Comment:
This was for a fix to fix an issue with getting the latest statuses when there is an overlap of date. Ken found an issue when 

Statement Paid --> Reverse Statement --> Statement Paid --> Reverse Statement (would error here because it would find a previous state)